docs/diagrams/EditModuleSequenceDiagram.puml

```diff
@@ -0,0 +1,122 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":EditModuleCommand" as EditModuleCommand LOGIC_COLOR
+participant "r:CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "m:Model" as Model MODEL_COLOR
+participant ":TrackIter" as TrackIter MODEL_COLOR
+end box
+
+[-> EditModuleCommand : execute(m)
+activate EditModuleCommand
+
+'getModule
+EditModuleCommand -> Model : getModule(code)
+activate Model
+Model -> TrackIter : getModule(code)
+TrackIter --> Model : moduleToEdit
+Model --> EditModuleCommand : moduleToEdit
+deactivate Model
+'end
+
+'createEditedModule
+EditModuleCommand -> EditModuleCommand : createEditedModule(moduleToEdit, editModuleDescriptor)
+activate EditModuleCommand
+EditModuleCommand --> EditModuleCommand : editedModule
+deactivate EditModuleCommand
+'end
+
+alt editedModule.code != moduleToEdit.code
+'getModuleTask
+EditModuleCommand -> Model : getModuleTasks(code)
+activate Model
+Model -> TrackIter : updateFilteredTaskList(pTask)
+Model -> TrackIter : getFilteredTaskList(pTask)
+note right
+pTask: predicate to only shows tasks
+that are associated with the module
+end note
+TrackIter -> Model : tasksToEdit
+Model --> EditModuleCommand : tasksToEdit
+deactivate Model
+'end
+
+'edit tasks
+loop task in tasksToEdit
+    EditModuleCommand -> Model : editTask(task)
+    activate Model
+    Model -> TrackIter : editTask(task)
+deactivate Model
+end
+'end
```

Comment:
The way this is displayed in the diagram looks a bit odd. The Arrow doesnt seem to point correctly. Also, the return arrow for tasksToEdit and lessonstoEdit is not a dotted arrow

Reply:
Ohhh yeahhh. Okay lemme see how can I fix it

Reply:
Fixed! Can you review it again? Thanks!